lib/Parse/ParseStmt.cpp

```diff
@@ -517,7 +517,7 @@ static ParserResult<Stmt> recoverFromInvalidCase(Parser &P) {
   assert(P.Tok.is(tok::kw_case) || P.Tok.is(tok::kw_default)
```

Comment:
I think you also need to adjust the doc comment `consuming up to the next ':' or '}'`